src/Microsoft.Identity.Web.TokenCache/Distributed/MsalDistributedTokenCacheAdapter.cs

```diff
@@ -263,14 +263,29 @@ protected override async Task WriteCacheBytesAsync(string cacheKey, byte[] bytes
                 SlidingExpiration = _distributedCacheOptions.SlidingExpiration,
             };
 
-            await L2OperationWithRetryOnFailureAsync(
-            write,
-            (cacheKey) => _distributedCache.SetAsync(
-                cacheKey,
-                bytes,
-                distributedCacheEntryOptions,
-                cacheSerializerHints?.CancellationToken ?? CancellationToken.None),
-            cacheKey).Measure().ConfigureAwait(false);
+            if (_distributedCacheOptions.DisableL1Cache)
```

Comment:
Would we want to have an option to disable the L2 fire and forget behavior?

Reply:
added. thanks.